### PR TITLE
Lots of Documentation Improvements

### PR DESCRIPTION
**Pull request by robscott**
This PR represents some big changes to the current rok8s-scripts documentation. Here are the biggest changes:

- Moving most documentation to `/docs` to allow for GitHub pages deployment once this is merged in (and also just more structured documentation)
- Consolidating CI examples for Jenkins and Circle into the same `CI` example, adding Bitbucket Pipelines and GitLab CI examples to that
- Removing references to older components that are no longer maintained (primarily anything related to minikube)
- Lots of more thorough documentation, especially related to secrets management and Helm

### CODE REVIEW

File: docs/docker.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+This will attempt to push the image with the following tags:
+```
+"${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_SHA1}"
+"${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_REF}"
+"${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:build_${CI_BUILD_NUM}"
+```
+
+An important note here is that `$CI_REF` defaults to `$CI_BRANCH`, enabling caching in the build step.
+
+## Change Detection
```
Comment (sudermanjr) [inline docs/docker.md]: I created this feature and I forgot it existed.  Nice work.

File: docs/helm.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+## Initial Project Structure
+
+All rok8s-scripts configuration lives in a `deploy` directory at the root of your project by default. In this example, we have a simple Python app with a Dockerfile in place.
+
+```plaintext
+app-dir/
+├── deploy/
+│   ├── build.config                         // rok8s-scripts build configs
+│   ├── charts/                              // All HELM chart configuration for the app
```
Comment (sudermanjr) [inline docs/helm.md]: My standard setup has usually been to leave charts in the root of the repo and reference them there.  Is this our normal recommendation to leave them in the deploy directory?

Comment (sudermanjr) [reply docs/helm.md]: It doesn't actually matter all that much, I was just curious.

Comment (robscott) [reply docs/helm.md]: That's interesting, I've always had them in a charts directory, but I guess there's really no need for that

**Pull request merged**